scripts/build/Platform/Windows/env_windows.cmd

```diff
@@ -22,21 +21,21 @@ IF NOT "%COMMAND_CWD%"=="" (
     CD %COMMAND_CWD%
 )
 
-REM Jenkins reports MSB8029 when TMP/TEMP is not defined, define a dummy folder
-IF NOT "%TMP%"=="" (
-    IF NOT "%WORKSPACE_TMP%"=="" (
-        SET TMP=%WORKSPACE_TMP%
-        SET TEMP=%WORKSPACE_TMP%
+ENDLOCAL
+
+REM Jenkins does not defined TMP
+IF "%TMP%"=="" (
+    IF "%WORKSPACE%"=="" (
+        SET TMP=%APPDATA%\Local\Temp
+        SET TEMP=%APPDATA%\Local\Temp
     ) ELSE (
-        SET TMP=%cd%/temp
-        SET TEMP=%cd%/temp
+        SET TMP=%WORKSPACE%\Temp
+        SET TEMP=%WORKSPACE%\Temp
     )
 )
-IF NOT EXIST "!TMP!" (
-    MKDIR "!TMP!"
-)
 
 EXIT /b 0
 
 :error
+ENDLOCAL
```

Comment:
Maybe we should put a comment here to explain what happens when we don't do this.

Reply:
sure